packages/websocket-plugin/src/websocket-handler.ts

```diff
@@ -51,18 +54,29 @@ export class WebSocketHandler {
     this.setupActionsListeners();
   }
 
-  private setupActionsListeners(): void {
-    this.actions$.pipe(ofActionDispatched(ConnectWebSocket)).subscribe(({ payload }) => {
-      this.connect(payload);
-    });
-
-    this.actions$.pipe(ofActionDispatched(DisconnectWebSocket)).subscribe(() => {
-      this.disconnect();
-    });
+  ngOnDestroy(): void {
+    this.closeConnection();
+    this.subscription.unsubscribe();
+  }
 
-    this.actions$.pipe(ofActionDispatched(SendWebSocketMessage)).subscribe(({ payload }) => {
-      this.send(payload);
-    });
+  private setupActionsListeners(): void {
+    this.subscription.add(
+      this.actions$.pipe(ofActionDispatched(ConnectWebSocket)).subscribe(({ payload }) => {
+        this.connect(payload);
+      })
+    );
+
+    this.subscription.add(
+      this.actions$.pipe(ofActionDispatched(DisconnectWebSocket)).subscribe(() => {
+        this.disconnect();
+      })
+    );
+
+    this.subscription.add(
+      this.actions$.pipe(ofActionDispatched(SendWebSocketMessage)).subscribe(({ payload }) => {
+        this.send(payload);
+      })
+    );
   }
```

Comment:
what do you think about?

```ts
private unsubscribe$: Subject<void> = new Subject();

public ngOnDestroy() {
  this.closeConnection();
  this.unsubscribe.next();
  this.unsubscribe.complete();
}

private setupActionsListeners(): void {
    this.ensureDispatchedActionStreamBy(ConnectWebSocket)
         .pipe(takeUntil(unsubscribe$))
         .subscribe(({ payload }) => this.connect(payload))

    this.ensureDispatchedActionStreamBy(SendWebSocketMessage)
         .pipe(takeUntil(unsubscribe$))
         .subscribe(({ payload }) => this.send(payload))

    this.ensureDispatchedActionStreamBy(DisconnectWebSocket)
       .pipe(takeUntil(unsubscribe$))
       .subscribe(() => this.disconnect())
}

private ensureDispatchedActionStreamBy<T>(action: ActionType): Observable<T> {
   return this.actions$.pipe(ofActionDispatched(action));
}



```

Reply:
Not sure if there is a difference between the current implementation.

Reply:
up to you)